tests/test_sequence_validation.py

```diff
@@ -152,51 +152,60 @@ def test_read_cleaning_trim_functions_lowercase(Tabletype, reads):
     for read in reads:
         x.consume(read.cleaned_seq)       # consume cleaned_seq
 
+    # all of these functions will fail to do anything, b/c lowercase != valid
+    # BUT they will not raise an exception, either.
+
     s = "caggcgcccaccaccgtgccctccaacctgatggt"
     _, where = x.trim_on_abundance(s, 1)
-    assert where == 35                    # in future, should be '0'
+    assert where == 0
 
-    _, where = x.trim_below_abundance(s, 2)
-    assert where == 35                    # in future, should be ?? @CTB
+    _, where = x.trim_below_abundance(s, 0)
+    print(x.get_kmer_counts(s))
+    assert where == 35                    # stays at 35 (abunds all == 0)
 
     posns = x.find_spectral_error_positions(s, 1)
-    assert posns == []                    # in future, should do same
+    assert posns == []
 
 
-def test_read_cleaning_trim_functions_N(Tabletype, reads):
+def test_read_cleaning_trim_functions_N(Countingtype, reads):
     # read this in using "approved good" behavior w/cleaned_seq
-    x = Tabletype(8, PRIMES_1m)
+    x = Countingtype(8, PRIMES_1m)
     for read in reads:
         x.consume(read.cleaned_seq)       # consume cleaned_seq
 
     s = "ACTGGGCGTAGNCGGTGTCCTCATCGGCACCAGC"
     _, where = x.trim_on_abundance(s, 1)
-    assert where == 0                     # in future, should be ??
+    assert where == 11
 
     _, where = x.trim_below_abundance(s, 2)
-    assert where == 0                     # in future, should be ??
+    assert where == 34
 
-    with pytest.raises(ValueError):
-        posns = x.find_spectral_error_positions(s, 1)
-    # assert posns == []                    # in future, should return []
+    posns = x.find_spectral_error_positions(s, 1)
+    assert posns == [11]
 
 
-def test_read_cleaning_trim_functions_bad_dna(Tabletype, reads):
+def test_read_cleaning_trim_functions_bad_dna(Countingtype, reads):
     # read this in using "approved good" behavior w/cleaned_seq
-    x = Tabletype(8, PRIMES_1m)
+    x = Countingtype(8, PRIMES_1m)
     for read in reads:
         x.consume(read.cleaned_seq)       # consume cleaned_seq
 
+    # the precise behavior of these functions is all *undefined*
+    # because different hash functions do different things with
+    # non-ACTG characters.  So all we want to do is verify that the
+    # functions execute w/o error on the k-mers before the "bad" DNA,
+    # and don't return positions in the "good" DNA.
```

Comment:
The behavior of the hash functions on ACTG should all be the same with respect to "good" sequences, and there should be no errors in there to trim at.